pom.xml

```diff
@@ -628,7 +628,7 @@
         <spark331.version>3.3.1</spark331.version>
         <spark332.version>3.3.2</spark332.version>
         <spark333.version>3.3.3-SNAPSHOT</spark333.version>
-        <spark340.version>3.4.0-SNAPSHOT</spark340.version>
+        <spark340.version>3.4.0</spark340.version>
```

Comment:
We also need to update the shim to stop accepting 3.4.0-SNAPSHOT and update shuffle docs per @tgravescs 's comment [here](https://github.com/NVIDIA/spark-rapids/pull/8105#issuecomment-1508946247).

Reply:
See @pxLi's comment here: https://github.com/NVIDIA/spark-rapids/pull/8105#pullrequestreview-1384824052

I discussed this with him after he brought this topic up first. The issue right now is that 3.4.0 build still doesn't pass testing. While this would unblock merging PRs, moving it to `noSnapshot.buildvers`  would currently block nightly from what I understand. If this is not true, I will update `noSnapshot.buildvers` as requested.

Reply:
Totally agree spark340 is not ready to be added to noSnapshot.buildvers.